### PR TITLE
Dispatch domain events in WebsiteBundleBundle

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR adjusts the AnalyticsManager to dispatch domain events.

### CODE REVIEW

**Comment by alexander-schranz** [comment]
Seems like there was a new release of the php setup CI action and now it is failing. I created a issue there not sure where the problem is: https://github.com/shivammathur/setup-php/issues/448

File: src/Sulu/Bundle/WebsiteBundle/Analytics/AnalyticsManager.php [hunk 1]
```diff
@@ -66,11 +77,14 @@ public function find($id)
 
     public function create($webspaceKey, $data)
     {
-        $entity = new Analytics();
+        /** @var AnalyticsInterface $entity */
+        $entity = $this->analyticsRepository->createNew();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/WebsiteBundle/Analytics/AnalyticsManager.php:81]: Missing in the `createNew` in the `AnalyticsRepositoryInterface`.

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/WebsiteBundle/Analytics/AnalyticsManager.php (hunk 1 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/WebsiteBundle/Analytics/AnalyticsManager.php:81]: derived from `RepositoryInterface` see https://github.com/sulu/sulu/pull/5968/files#diff-77a097117b1e5256caea7c780d12fea32d121876b6c841c6465226b353577869R16

File: src/Sulu/Bundle/WebsiteBundle/Entity/AnalyticsRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\Entity;
+
+use Sulu\Component\Persistence\Repository\RepositoryInterface;
+
+interface AnalyticsRepositoryInterface extends RepositoryInterface
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/WebsiteBundle/Entity/AnalyticsRepositoryInterface.php]: I would not extend in the `AnalyticsRepositoryInterface` from `RepositoryInterface` if needed the implementation can implement `AnalyticsRepository` class can implement `RepositoryInterface` but the `AnalyticsRepositoryInterface` should stay for itself and not implementing infrastructure interfaces

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

File: phpstan-baseline.neon [hunk 3]
```diff
@@ -48770,51 +48765,11 @@ parameters:
...
-
-		-
-			message: "#^Property Sulu\\\\Bundle\\\\WebsiteBundle\\\\Entity\\\\Analytics\\:\\:\\$content type mapping mismatch\\: database can contain array but property expects string\\.$#"
-			count: 1
-			path: src/Sulu/Bundle/WebsiteBundle/Entity/Analytics.php
-
-		-
-			message: "#^Property Sulu\\\\Bundle\\\\WebsiteBundle\\\\Entity\\\\Analytics\\:\\:\\$content type mapping mismatch\\: property can contain string but database expects array\\.$#"
-			count: 1
-			path: src/Sulu/Bundle/WebsiteBundle/Entity/Analytics.php
```
Comment (niklasnatter) [inline phpstan-baseline.neon]: 👏

File: src/Sulu/Bundle/WebsiteBundle/DependencyInjection/SuluWebsiteExtension.php [hunk 4]
```diff
@@ -149,5 +152,7 @@ public function load(array $configs, ContainerBuilder $container)
                 $loader->load('exception_controller.xml');
             }
         }
+
+        $this->configurePersistence($config['objects'], $container);
```
Comment (niklasnatter) [inline src/Sulu/Bundle/WebsiteBundle/DependencyInjection/SuluWebsiteExtension.php:157]: would be nice to add an alias for the repository for autowiring:
https://github.com/sulu/sulu/blob/2.x/src/Sulu/Bundle/CategoryBundle/DependencyInjection/SuluCategoryExtension.php#L47-L54

File: src/Sulu/Bundle/WebsiteBundle/Resources/config/analytics.xml [hunk 5]
```diff
@@ -24,16 +16,17 @@
         <service id="sulu_website.analytics.manager"
                  class="Sulu\Bundle\WebsiteBundle\Analytics\AnalyticsManager" public="true">
             <argument type="service" id="doctrine.orm.entity_manager"/>
-            <argument type="service" id="sulu_website.analytics.repository"/>
+            <argument type="service" id="sulu.repository.analytics"/>
```
Comment (niklasnatter) [inline src/Sulu/Bundle/WebsiteBundle/Resources/config/analytics.xml:18]: think we need to add this change to the `UPGRADE.md`

Comment (wachterjohannes) [reply src/Sulu/Bundle/WebsiteBundle/Resources/config/analytics.xml:18]: @nnatter have added aliases for the old id and for the interface

File: src/Sulu/Bundle/WebsiteBundle/Resources/config/analytics.xml [hunk 6]
```diff
@@ -3,16 +3,8 @@
...
-            <factory service="doctrine.orm.entity_manager" method="getRepository"/>
-
-            <argument type="string">SuluWebsiteBundle:Analytics</argument>
-
-            <call method="setEnvironment">
-                <argument>%kernel.environment%</argument>
-            </call>
-        </service>
+        <service id="sulu_website.analytics.repository" alias="sulu.repository.analytics"/>
+        <service id="Sulu\Bundle\WebsiteBundle\Entity\AnalyticsRepositoryInterface" alias="sulu.repository.analytics"/>
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/WebsiteBundle/Resources/config/analytics.xml]: @nnatter see here

Comment (niklasnatter) [reply src/Sulu/Bundle/WebsiteBundle/Resources/config/analytics.xml]: would you mind to the `SuluWebsiteExtension` to be consistent with the other bundles? i am afraid that people (like me 😅) will copy the code from the extension class and forget abput the alias in the xml file

**Pull request merged**